pkgs/development/libraries/SDL_compat/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, SDL2
+, libGLSupported ? lib.elem stdenv.hostPlatform.system lib.platforms.mesaPlatforms
+, openglSupport ? libGLSupported
+, libGL
+, libGLU
+}:
+
+let
+  inherit (lib) optionals;
```

Comment:
It is less maintenance to just write out ``lib.`` instead of maintaining this list.

Reply:
> For any particular reason?

Treewide commits are always done in one and not spread across 17 commits.

> It's far easier to undo something when each change is its own commit and it's highly unlikely we would want to mass undo all of them.

This is one change. You can't revert one of the commits without triggering an eval error. Also it neither a risk nor breaking change.

Reply:
> You can't revert one of the commits without triggering an eval error

Sure you can. These commits are only really needed for when you do `SDL = SDL_compat;` - otherwise they make no difference.

> Treewide commits are always done in one and not spread across 17 commits.

But for what reason? There are pros to keeping them separate. The only con would be if they all need to be reverted in one go, which is highly unlikely.

Reply:
> The only con would be if they all need to be reverted in one go, which is highly unlikely.

If SDL_compat does not work on some system combination which you didn't test but is highly popular then they all would need to be reverted if the problem is not easily fixable. I don't believe that this is highly unlikely but also not very likely. 

> But for what reason?

That you know that the change was treewide and not necessarily every application was tested because it should universally work. It makes reverting, cherry-picking and iterating on it easier and you can't easily miss one commit when reverting. ofborg is not perfect and will not catch every error.

Reply:
> If SDL_compat does not work on some system combination which you didn’t test but is highly popular then they all would need to be reverted if the problem is not easily fixable. I don’t believe that this is highly unlikely but also not very likely. 

Let me try to explain this in a different way. 

1. This tree-wide change should be done irrespectively of us choosing to include SDL_compat or not, let alone switch SDL over to use SDL_compat.
2. Some of these changes might be wrong. Not all, some. One example is a reference to SDL.out which has been replaced by getLib SDL.
3. We definitely don’t want to undo them all. Changing SDL.dev to getDev SDL is going to be fine and is in no way related to the change above.
4. These changes can be done in isolation. No eval errors as a consequence of not doing them all.

Maybe it’s my use of the word tree-wide that caused the confusion. 

 > That you know that the change was treewide and not necessarily every application was tested because it should universally work. It makes reverting, cherry-picking and iterating on it easier and you can’t easily miss one commit when reverting. ofborg is not perfect and will not catch every error. 

But we don’t want to revert them all. You would only want to revert the problematic ones (if any).